src/optimize_embedding_potential.F

```diff
@@ -2518,8 +2582,368 @@ SUBROUTINE Leeuwen_Baerends_potential_update(pw_env, embed_pot, spin_embed_pot,
+      REAL(kind=dp), DIMENSION(:, :, :), POINTER         :: rhoa, rhob
+      TYPE(pw_env_type), POINTER                         :: pw_env
+      TYPE(pw_p_type), DIMENSION(:), POINTER             :: rho_g, tau
+      TYPE(pw_pool_type), POINTER                        :: auxbas_pw_pool
+      TYPE(section_vals_type), POINTER                   :: input, xc_section
+      TYPE(xc_rho_cflags_type)                           :: needs
+      TYPE(xc_rho_set_type), POINTER                     :: rho_set
+
+      one_4 = 0.25_dp
+      one_8 = 0.125_dp
+
+      rho_cutoff = EPSILON(0.0_dp)
+
+      nspins = SIZE(rho_r)
+
+      NULLIFY (rho_set, xc_section)
+
+      CALL get_qs_env(qs_env=qs_env, &
+                      pw_env=pw_env, &
+                      input=input)
+
+      ! Get plane waves pool
+      CALL pw_env_get(pw_env, auxbas_pw_pool=auxbas_pw_pool)
+
+      ! get some of the grids ready
+      NULLIFY (rho_g)
+      ALLOCATE (rho_g(nspins))
+      DO i_spin = 1, nspins
+         NULLIFY (rho_g(i_spin)%pw)
+         CALL pw_pool_create_pw(auxbas_pw_pool, rho_g(i_spin)%pw, &
+                                use_data=COMPLEXDATA1D, &
+                                in_space=RECIPROCALSPACE)
+         CALL pw_transfer(rho_r(i_spin)%pw, rho_g(i_spin)%pw)
+      ENDDO
+
+      xc_section => section_vals_get_subs_vals(input, "DFT%XC")
+
+      CALL xc_rho_set_create(rho_set, &
+                             rho_r(1)%pw%pw_grid%bounds_local, &
+                             rho_cutoff=section_get_rval(xc_section, "density_cutoff"), &
+                             drho_cutoff=section_get_rval(xc_section, "gradient_cutoff"), &
+                             tau_cutoff=section_get_rval(xc_section, "tau_cutoff"))
+
+      CALL xc_rho_cflags_setall(needs, .FALSE.)
+
+      IF (nspins .EQ. 2) THEN
+         needs%rho_spin = .TRUE.
+         needs%norm_drho_spin = .TRUE.
+         needs%laplace_rho_spin = .TRUE.
+      ELSE
+         needs%rho = .TRUE.
+         needs%norm_drho = .TRUE.
+         needs%laplace_rho = .TRUE.
+      ENDIF
+
+      CALL xc_rho_set_update(rho_set, rho_r, rho_g, tau, needs, &
+                             section_get_ival(xc_section, "XC_GRID%XC_DERIV"), &
+                             section_get_ival(xc_section, "XC_GRID%XC_SMOOTH_RHO"), &
+                             auxbas_pw_pool)
+
+      CALL section_vals_val_get(xc_section, "DENSITY_CUTOFF", &
+                                r_val=rho_cutoff)
+      CALL section_vals_val_get(xc_section, "DENSITY_SMOOTH_CUTOFF_RANGE", &
+                                r_val=density_smooth_cut_range)
+
+      lb(1:3) = rho_r(1)%pw%pw_grid%bounds_local(1, 1:3)
+      ub(1:3) = rho_r(1)%pw%pw_grid%bounds_local(2, 1:3)
+
+      IF (nspins .EQ. 2) THEN
+!$OMP    PARALLEL DO DEFAULT(NONE) &
+!$OMP                PRIVATE(i,j,k, my_rho) &
+!$OMP                SHARED(v_w, rho_r, rho_set, lb, ub, rho_cutoff, one_4, one_8)
```

Comment:
I think, you can keep `one_4` and `one_8` as `PARAMETERS`. Simply remove them from the `OMP SHARED` block.